solr/core/src/java/org/apache/solr/search/ValueSourceParser.java

```diff
@@ -1169,9 +1174,9 @@ public ValueSource parse(FunctionQParser fp) throws SyntaxError {
           @Override
           public ValueSource parse(FunctionQParser fp) throws SyntaxError {
             if (fp.sp.peek() == QueryParsing.LOCALPARAM_START.charAt(0)) {
-              return new UniqueBlockQueryAgg(fp.parseNestedQuery());
+              return new UniqueBlockQueryAgg(fp.parseNestedQuery(), UniqueAgg.DEFAULT_NUM_VALS_EXPLICIT);
             }
-            return new UniqueBlockFieldAgg(fp.parseArg());
+            return new UniqueBlockFieldAgg(fp.parseArg(), UniqueAgg.DEFAULT_NUM_VALS_EXPLICIT);
```

Comment:
curious why we don't support specifying `numValsExplicit` arg here? Is it just because we don't use `UniqueBlock*`?

Reply:
Mainly because we don't use it. But certainly could add support -- especially if this were upstreamed.